### ISSUE TITLE
Unhandled exception - compile-time "if" without braces

### ISSUE THREAD

**Issue by curt-w** [postsharp/Metalama#20]
Low priority in my opinion, just reporting as an FYI. Seems an unhandled exception occurs during build if compile time code contains an "if" statement without curly braces.

I've attached the exception log that is thrown and provided examples below.

[exception-e145db41-6543-4668-a22b-bde493a12723.log](https://github.com/postsharp/Metalama/files/8840468/exception-e145db41-6543-4668-a22b-bde493a12723.log)

```
// this builds successfully
if (someCondition)
{
    someValue = 0;
}

// this single line statement fails build
if (someCondition)
    someValue = 0;

// this one liner also fails, same as above I assume but not sure if it compiles differently...
if (someCondition) someValue = 0;
```

**Comment by prochan2** [comment]
Thank you for reporting this issue. I wasn't able to reproduce it, unfortunately. May I ask you for a pair of aspect code and target code, which reproduces the issue? An easy way to reproduce and share the reproduction is to reproduce it using [try.metalama.net](https://try.metalama.net) and copy the link created using the "Share" button there.

**Comment by curt-w** [comment]
Looks like issue might be isolated to compile-time code. I've noticed the error only occurs in situations like such:

```
if (someCondition)
    meta.Proceed();
```

Please see the following [example](https://try.metalama.net/#N4IghiBcIPQwBAFQBYEsDO8PzAO3gPYAOAprgLToECuATgMYnwCyJALmADZgC2Y8JAB68inEgDp4AZRJNkbNkXSQ4Ac1Rtk1AEbj6BHjCIF0bdMjC0iMVh259xUkWMwAzArXg8PEgDq5-anRUXFUWdi5eMHEAMVpeEgB3DwBrcQBBdFJ6MwBuQODQ8Lso2PieJNTYsG1aVHp0fNwgkLDbSIc4hOTaNIBhAgATEib-fxC2ElpcLnh6bnRMGJq6+nhIeAAFWgIAKxIc5dr6-2Ax-HhLoh1OesIANym64fh7glRB+HSK3EHtvYObAAFABJf77HLfMjDTwJX5TACU-kulzOFxRKPulngbAAnqR0AAlA4kVCPTwAXhwPxh4gA6hpkECiDtdvAKQA+eAsvbiRD4kjoZEYkWisXiiUY+nIKYkIF40jsrkg9D80gAUVu6m0YnlAoRBqa4qxnh5pFobFQguJjDJU3Z1OhU3pjOZrKV3NZfIFQvRkv9AfF0tlesVnPgKrVJE1qG1uoVJANwsDKdTjhIYhyzDwuNDTHDCYynE4-3NlsFwdocrNHpVpameJjcerOyIBsN53FcATRJJdtoGUGg0y2TYAB4+hZQiREPF6ClWlH0CPARygQijWKzQ2rb3beTB8OsoCx6WAPLk54kNcb87J+AAX07KJZZLAk3g2gIBE4EdVAqbVAdTlEEADkEkGKMcX1e80XFKs2DofBCxVPMCFcIFEEFNg+gWdAOz9J8Aj9V8sQ-L8fz-esLVxQDgNBai8U9YgG1xJE-TgsUEKQnFaGoEZ7yI-whNwa4dTueYwEWLZWwvJ4PiYDY5NoK8YitThBjPWhGNxFcclOe8xNuNYCEvBT4EGXEZh4eoAH54GUq8dNg+8MVUdhXJRTjJW46YvAicR-kYEhBnXPkCAAVSIc1103UUiPFdAPL9DFvIlOBUFceAgQARnZKkcvY1N-I4QKdmC0LbxSlEEpq4SQAAGhAIgoFgBAUGwbA8EIUgKCoOhGGKDp+CEZwJGkWR4HkRRlDURkdD0AwjBMMwLCsGwInsaInB4URBXgdxPG8KtxHOIyJLwpBsNwqTfTS861lMOoilLHKAGYACZ4Bc6rLnctg4pFJKAcE4TzgmKYZl-STpP+VRygMki6jIphTHfO43g+FgwBCdcfuNbFEg+f6HVwJIrtMG7Fli59RSJwZ-rK4h3q+qlfHAMAwHZ0ZfvgAZcCoMR6TqSYABkQjlenGdez6qoxESHxAB8gA).

Currently if you run it, you will notice the aspect is correctly capitalizing the output of the property. If you uncomment the "if (1 == 1)" that precedes the "meta.Proceed()" call in the setter, you will notice the aspect is no longer applied to the property. While no "exception" is being thrown, I believe the error might be occurring under the scenes which is resulting in the aspect no longer being applied. When I run this in my VS2022 environment, I receive the build error noted earlier.

**Comment by addabis** [comment]
Thanks for more information.

I've successfully reproduced the problem.

We will be working on a fix and notify you as soon as it is released.

**Comment by gfraiteur** [comment]
Solved in 0.5.31-preview.
